spec/javascripts/social_networking/profile/profile-answer-controller_spec.js

```diff
@@ -0,0 +1,161 @@
+describe('ProfileAnswerCtrl', function() {
+  var $q, $rootScope, $window, answerText, controller, profileId, profileAnswers, questionId, questions;
+
+  beforeEach(module('socialNetworking.controllers'));
+
+  beforeEach(inject(function($controller, _$rootScope_, _$q_, _$window_) {
+    answerText = 'I love it!';
+    controller = $controller;
+    profileId = 1038927012;
+    questionId = 781294868;
+    $rootScope = _$rootScope_;
+    $q = _$q_;
+    $window = _$window_;
+    questions = [
+      {
+        id: questionId,
+        question_text: 'What are your hobbies?'
+      }
+    ];
+  }));
+
+  describe('#ifAnswered', function() {
+    beforeEach(function() {
+      controller = controller('ProfileAnswerCtrl', {
+        ProfileAnswers: function() {}
+      });
+    });
+
+    it('returns `true` if the participant has answered at least one profile question', function() {
+      controller.answerModels = {}
+      controller.answerModels[questionId] = {
+        answer_text: answerText,
+        profile_id: profileId,
+        profile_question_id: questionId
+      };
+
+      expect(controller.ifAnswered(questions)).toBe(true);
+    });
+
+    it('returns `false` if the participant has not answered at least one profile question', function() {
+      controller.answerModels = {};
+
+      expect(controller.ifAnswered(questions)).toBe(false);
+    });
+  });
+
+  describe('#save', function() {
+    beforeEach(function() {
+      profileAnswers = jasmine.createSpyObj('profileAnswers', ['create', 'getOne']);
+
+      profileAnswers.getOne.and.callFake(function() {
+        var deferred, promise, response;
+
+        response = {
+          answer_text: answerText,
+          profile_id: profileId,
+          profile_question_id: questionId
+        }
+
+        deferred = $q.defer();
+        promise = deferred.promise;
+        deferred.resolve(response);
+        return promise;
+      });
+
+      profileAnswers.create();
+    });
+
+    describe('when saving succeeds', function() {
+      beforeEach(function() {
+        profileAnswers.create.and.callFake(function() {
+          var deferred, promise;
+
+          deferred = $q.defer();
+          promise = deferred.promise;
+          deferred.resolve();
+          return promise;
+        });
+
+        controller = controller('ProfileAnswerCtrl', {
+          ProfileAnswers: profileAnswers
+        });
+
+        controller._answerStates = {
+          'foo': {}
+        };
+
+        spyOn(controller, 'getModel').and.callFake(function() {
```

Comment:
I realize that I am mocking a function of the object I am testing, but I don't see a way around this.